src/WingetCreateCLI/PromptHelper.cs

```diff
@@ -0,0 +1,270 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+
+namespace Microsoft.WingetCreateCLI
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.ComponentModel.DataAnnotations;
+    using System.Linq;
+    using System.Reflection;
+    using Microsoft.WingetCreateCLI.Logging;
+    using Microsoft.WingetCreateCLI.Properties;
+    using Microsoft.WingetCreateCore;
+    using Microsoft.WingetCreateCore.Common;
+    using Microsoft.WingetCreateCore.Models.Installer;
+    using Newtonsoft.Json;
+    using Sharprompt;
+
+    /// <summary>
+    /// Provides functionality for prompting the user for input to obtain field values for a manifest.
+    /// </summary>
+    public static class PromptHelper
+    {
+        /// <summary>
+        /// List of strings representing the optional fields that should not be editable.
+        /// </summary>
+        private static readonly string[] NonEditableOptionalFields = new[]
+        {
+            nameof(InstallerType),
+            nameof(InstallerManifest.Channel),
+            nameof(Installer.InstallerUrl),
+            nameof(Installer.InstallerSha256),
+            nameof(Installer.SignatureSha256),
+            nameof(Installer.ProductCode),
+            nameof(Installer.PackageFamilyName),
+            nameof(Installer.AdditionalProperties),
+        };
+
+        /// <summary>
+        /// Displays all properties of a model as a navigational menu selection with some additional filtering logic based on InstallerType.
+        /// </summary>
+        /// <typeparam name="T">Model type.</typeparam>
+        /// <param name="model">Instance of object model.</param>
+        /// <param name="exitMenuWord">Exit keyword to be shown to the user to exit the navigational menu.</param>
+        public static void DisplayPropertiesAsMenuSelection<T>(T model, string exitMenuWord)
```

Comment:
This name implies that it merely displays the properties, but it also allows the user to change the prop. I think something like PromptPropertiesWithMenu or something like that would be more apt

Reply:
I like that better, changed to suggestion